pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageCrypto.java

```diff
@@ -511,7 +529,8 @@ private ByteBuf decryptDataKeyAndData(MessageMetadata msgMetadata, ByteBuf paylo
                     break;
                 }
             } catch (Exception e) {
-                // First time, entry won't be present in cache 
+                // First time, entry won't be present in cache
+                log.debug("Failed to decrypt data or data key is not in cache. Will attempt to refresh");
```

Comment:
Please add more contextual info (topic, etc..)

Reply:
Exception..

Reply:
Added logCtx, so we can log topic/producername/id for producer and topic/subscription for consumer.